app/features/project/project_controller.rb

```diff
@@ -171,10 +170,22 @@ class ProjectController < AuthenticatedControllerBase
         platform: lane.split(" ").last,
         lane: lane.split(" ").first,
         # TODO: Until we make a proper interface to attach JobTriggers to a Project, let's add a manual one for the selected branch.
-        job_triggers: [FastlaneCI::ManualJobTrigger.new(branch: branch)]
+        job_triggers: (trigger.nil? ? [] : [trigger])
       )
 
       if !project.nil?
+        # Do this so we trigger the clone of the repo.
+        # TODO: Do this wherever it should be done, as we must redirect
+        # to the project details only when this task is finished.
+        repo = GitRepo.new(
+          git_config: repo_config,
+          provider_credential: provider_credential,
+          local_folder: project.local_repo_path,
+          async_start: false
+        )
+
+        repo.checkout_branch(branch: branch, use_global_git_mutex: false)
```

Comment:
fyi you almost always want to use `use_global_git_mutex: true`
The only time you don't want to use it is when you're already using the `GitRepo.git_action_queue`, since the serial queue isn't reentrant 👍